src/restic/backend/b2/b2.go

```diff
@@ -0,0 +1,288 @@
+package b2
+
+import (
+	"io"
+	"path"
+	"restic"
+	"strings"
+
+	"restic/debug"
+	"restic/errors"
+
+	blazerb2 "github.com/kurin/blazer/b2"
+
+	"golang.org/x/net/context"
+)
+
+// b2 is a backend which stores the data on a B2 endpoint.
+type b2 struct {
+	client     *blazerb2.Client
+	bucket     *blazerb2.Bucket
+	bucketName string
+	prefix     string
+	context    context.Context
+}
+
+// Open opens the B2 backend. The bucket is created if it does not exist yet.
+func Open(cfg Config) (restic.Backend, error) {
+	debug.Log("open, config %#v", cfg)
+
+	ctx := context.Background()
+
+	client, err := blazerb2.NewClient(ctx, cfg.AccountID, cfg.Key)
+	if err != nil {
+		return nil, errors.Wrap(err, "blazerb2.NewClient")
+	}
+
+	bucket, err := client.NewBucket(ctx, cfg.Bucket, blazerb2.Private)
+	if err != nil {
+		return nil, errors.Wrap(err, "blazerb2.NewBucket")
+	}
+
+	be := &b2{client: client, bucket: bucket, bucketName: cfg.Bucket, prefix: cfg.Prefix, context: ctx}
+
+	return be, nil
+}
+
+// Resolve the backend path for based on file type ane name.
+func (be *b2) b2path(t restic.FileType, name string) string {
+	if t == "key" && name == "config" {
+		t = restic.ConfigFile
+	}
+	if string(t) == restic.ConfigFile || name == "" {
+		return path.Join(be.prefix, string(t))
+	}
+	return path.Join(be.prefix, string(t), name)
+}
+
+// Location returns this backend's location (the bucket name).
+func (be *b2) Location() string {
+	return be.bucketName
+}
+
+// Load returns the data stored in the backend for h at the given offset
+// and saves it in p. Load has the same semantics as io.ReaderAt.
+func (be b2) Load(h restic.Handle, p []byte, off int64) (n int, err error) {
+	debug.Log("Load: %v, offset %v, len %v", h, off, len(p))
+	objName := be.b2path(h.Type, h.Name)
+
+	obj := be.bucket.Object(objName)
+	if err != nil {
+		debug.Log("  err %v", err)
+		return 0, errors.Wrap(err, "bucket.Object")
+	}
+
+	info, err := obj.Attrs(be.context)
+	if err != nil {
```

Comment:
This needs to check for `attrs.Status == b2.Uploaded`, otherwise the read might fail.